hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -248,15 +249,18 @@ public static HoodieWriteClient createHoodieClient(JavaSparkContext jssc, String
   }
 
   public static JavaRDD<WriteStatus> doWriteOperation(HoodieWriteClient client, JavaRDD<HoodieRecord> hoodieRecords,
-      String instantTime, String operation) throws HoodieException {
-    if (operation.equals(DataSourceWriteOptions.BULK_INSERT_OPERATION_OPT_VAL())) {
+      String instantTime, WriteOperationType operation) throws HoodieException {
```

Comment:
Also, since we have changed this to enum, can you also update DataSourceWriteOptions to reference this enum. It is good that both the enum values and those defined in DataSourceWriteOptions are consistent. So, we can do this without any backwards compatibility issues.

```
  val OPERATION_OPT_KEY = "hoodie.datasource.write.operation"
  val BULK_INSERT_OPERATION_OPT_VAL = WriteOperationType.BULK_INSERT.name
  val INSERT_OPERATION_OPT_VAL = WriteOperationType.INSERT.name
  val UPSERT_OPERATION_OPT_VAL = WriteOperationType.UPSERT.name
  val DELETE_OPERATION_OPT_VAL = WriteOperationType.DELETE.name
  val BOOTSTRAP_OPERATION_OPT_VAL = WriteOperationType.BOOTSTRAP.name
  val DEFAULT_OPERATION_OPT_VAL = UPSERT_OPERATION_OPT_VAL
```